torchci/lib/bot/retryBot.ts

```diff
@@ -54,27 +53,42 @@ function retryBot(app: Probot): void {
       // or a widespread outage that wouldn't be helped by retries
       return;
     }
+
+    // @ts-expect-error - we don't have types for these
+    let jobsDoesntFailStepsLike = (job, predicate: (step: any) => boolean) => {
+      return job.steps?.filter(
+        // @ts-expect-error
+        (step) =>
+          step.conclusion !== null &&
+          predicate(step) &&
+          FAILURE_CONCLUSIONS.includes(step.conclusion)
+      ).length === 0
+    }
+
     const shouldRetry = failedJobs.filter((job) => {
-      // always rerun lint
-      if (workflowName === "lint") {
-        return true;
-      }
       // if the job was cancelled, it was probably an infra error, so rerun
       if (job.conclusion === "cancelled") {
         return true;
       }
+
+      // rerun if the linter didn't fail on the actual linting steps
+      if (workflowName === "lint" &&
+        jobsDoesntFailStepsLike(job, step => step.name.toLowerCase().startsWith("run lint - "))){
```

Comment:
Is that why you need: https://github.com/pytorch/pytorch/pull/90705 ? Could it perhaps be done differently somehow? Like querying some extra property?

Reply:
Yeah, this was the reason for that PR. There isn't any existing property that offers this information, the renaming of certain steps is what adds that data